04_library_management_system.rb

```diff
@@ -0,0 +1,143 @@
+class Member
+  attr_accessor :name, :book_issued, :status
+
+  def initialize(name)
+    @name = name
+    @status = "No book taken"
+  end
+
+  def book_taken?
+    if @status == "No book taken"
+      return false
+    else
+      return true
+    end
+  end
+end
+
+class Book
+  attr_accessor :name, :issued_to, :status, :message
+
+  def initialize(name)
+    @name = name
+    @status = "New Book"
+  end
+
+  def issue_to_member(member)
+    if (@issued_to.nil? || book_available?)
+      if (!member.book_taken?)
+        member.status = "Book taken"
+        @issued_to = member
+        @status = "Not Available"
+        @message = "#{@name} issued to #{member.name}"
+      else
+        @message = "Please return earlier borrowed book first"
+      end
+    else
+      @message = "Sorry, #{@name} is not available. Please try another book"
+    end
+    return @message
```

Comment:
Should I remove this explicit `return` as well?

Reply:
Yes

Reply:
Can I remove the `return @message` entirely since `@message` is the last assigned statement in each method or it is better to keep it explicit

Reply:
We can remove it.